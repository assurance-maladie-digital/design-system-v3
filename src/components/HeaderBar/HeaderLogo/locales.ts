export const locales = {
	ariaLabel: 'Ouvrir le menu',
}
