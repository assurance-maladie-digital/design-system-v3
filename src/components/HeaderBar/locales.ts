export const locales = {
	homeAriaLabel: 'Logo de l\'Assurance Maladie, cliquez pour revenir Ã  l\'accueil',
}
