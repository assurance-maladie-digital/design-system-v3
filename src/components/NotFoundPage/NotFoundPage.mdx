import { Controls, Canvas, Meta, Source } from "@storybook/blocks";
import * as NotFoundPageStories from "./NotFoundPage.stories";

<Meta of={NotFoundPageStories} />

# NotFoundPage

Le composant `NotFoundPage` permet d'afficher une page 404.
Il est bas√© sur le composant `ErrorPage` du design system.

<Canvas of={NotFoundPageStories.Default} />