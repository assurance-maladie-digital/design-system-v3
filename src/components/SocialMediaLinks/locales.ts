export const locales = {
	followUs: 'Suivez-nous',
}
