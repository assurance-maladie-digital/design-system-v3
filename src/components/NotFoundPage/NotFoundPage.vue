<script setup lang="ts">
	import { onMounted, ref } from 'vue'
	import ErrorPage from '../ErrorPage/ErrorPage.vue'
	import { locales } from './locales'

	const SUPPORT_ID_PARAM_NAME = 'support_id'
	const supportId = ref<string | undefined>()

	onMounted(() => {
		const params = new URLSearchParams(document.location.search)
		let supportIdParam = params.get(SUPPORT_ID_PARAM_NAME)

		if (!supportIdParam) {
			return
		}

		supportId.value = supportIdParam.trim().match(/.{1,4}/g)?.join(' ')
	})
</script>

<template>
	<ErrorPage
		:code="locales.code"
		:page-title="locales.pageTitle"
		:message="locales.message"
	>
		<template
			v-if="supportId"
			#additional-content
		>
			<p class="mt-4">
				{{ locales.supportIdMessage }}

				<b>{{ supportId }}</b>
				.
			</p>
		</template>

		<template #illustration>
			<img
				src="./assets/not-found.svg"
				:alt="locales.pageTitle"
			>
		</template>
	</ErrorPage>
</template>

<style lang="scss" scoped>
img {
	max-height: 290px;
}
</style>
