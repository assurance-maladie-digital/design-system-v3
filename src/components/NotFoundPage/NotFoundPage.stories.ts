import type { Meta, StoryObj } from '@storybook/vue3'
import NotFoundPage from './NotFoundPage.vue'

const meta = {
	title: 'Templates/NotFoundPage',
	component: NotFoundPage,
	parameters: {
		layout: 'fullscreen',
	},
} satisfies Meta<typeof NotFoundPage>

export default meta

type Story = StoryObj<typeof NotFoundPage>

export const Default: Story = {
	parameters: {
		sourceCode: [
			{
				name: 'Template',
				code: `
				<template>
					<div style="padding: 20px; background: rgb(231, 236, 245)">
						<NotFoundPage />
					</div>
				</template>
				`,
			}, {
				name: 'Script',
				code: `
				<script setup lang="ts">
					import { NotFoundPage } from '@cnamts/synapse'
				</script>

				`,
			},
		],
	},
	decorators: [
		() => ({ template: '<div style="padding: 20px; background: rgb(231, 236, 245)"><story /></div>' }),
	],
}
