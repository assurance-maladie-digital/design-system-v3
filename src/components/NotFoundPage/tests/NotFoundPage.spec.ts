import { describe, it, expect } from 'vitest'
import { mount } from '@vue/test-utils'

import { vuetify } from '@tests/unit/setup'
import { locales } from '../locales'
import NotFoundPage from '../NotFoundPage.vue'

describe('NotFoundPage', () => {
	it('renders correctly', () => {
		const wrapper = mount(NotFoundPage, {
			global: {
				plugins: [vuetify],
			},
		})

		expect(wrapper.text()).toContain(locales.code)
		expect(wrapper.text()).toContain(locales.message)
		expect(wrapper.html()).toMatchSnapshot()
	})

	it('display the support ID if provided in the url', async () => {
		history.replaceState(
			{},
			'',
			'/not-found?support_id=1234567890123456789',
		)
		const wrapper = mount(NotFoundPage, {
			global: {
				plugins: [vuetify],
			},
		})

		await wrapper.vm.$nextTick()

		expect(wrapper.text()).toContain('1234 5678 9012 3456 789')
		expect(wrapper.html()).toMatchSnapshot()
	})
})
