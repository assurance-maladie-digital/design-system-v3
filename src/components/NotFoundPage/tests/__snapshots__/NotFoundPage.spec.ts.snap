// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NotFoundPage > display the support ID if provided in the url 1`] = `
"<div data-v-fb0b02fd="" data-v-8f2b5bad="" data-v-c9c5b952="" class="py-10 px-8 vd-page-container d-flex justify-center">
  <p data-v-fb0b02fd="">Container size: 960</p>
  <div data-v-fb0b02fd="" class="v-sheet v-theme--light bg-transparent" style="width: 960px;">
    <div data-v-8f2b5bad="" class="v-card v-theme--light v-card--density-default elevation-0 v-card--variant-elevated pa-6 pa-sm-16">
      <!---->
      <div class="v-card__loader">
        <div class="v-progress-linear v-theme--light v-locale--is-ltr" style="top: 0px; height: 0px; --v-progress-linear-height: 2px;" role="progressbar" aria-hidden="true" aria-valuemin="0" aria-valuemax="100">
          <!---->
          <div class="v-progress-linear__background" style="opacity: NaN;"></div>
          <div class="v-progress-linear__buffer" style="opacity: NaN; width: 0%;"></div>
          <transition-stub name="fade-transition" appear="false" persisted="false" css="true">
            <div class="v-progress-linear__indeterminate">
              <div class="v-progress-linear__indeterminate long"></div>
              <div class="v-progress-linear__indeterminate short"></div>
            </div>
          </transition-stub>
          <!---->
        </div>
      </div>
      <!---->
      <!---->
      <div data-v-8f2b5bad="" class="v-row max-width-none">
        <div data-v-8f2b5bad="" class="v-col-sm-6 v-col-12 order-last order-sm-first text-center text-sm-left d-flex flex-column justify-center align-sm-start">
          <div data-v-8f2b5bad="" class="vd-code text-primary mb-4"><span data-v-8f2b5bad="" class="d-sr-only">Code d'erreur&nbsp;: </span> 404</div>
          <h2 data-v-8f2b5bad="" class="mb-2 font-weight-bold text-h5 mb-4">Page non trouvée</h2>
          <p data-v-8f2b5bad="">Cette page n’existe pas ou a été déplacée.</p>
          <p data-v-c9c5b952="" class="mt-4">Votre identifiant de support est <b data-v-c9c5b952="">1234 5678 9012 3456 789</b> . </p>
          <!--v-if-->
        </div>
        <div data-v-8f2b5bad="" class="v-col-sm-6 v-col-12 d-flex align-center justify-center"><img data-v-c9c5b952="" src="/src/components/NotFoundPage/assets/not-found.svg" alt="Page non trouvée"></div>
      </div>
      <!---->
      <!----><span class="v-card__underlay"></span>
    </div>
  </div>
</div>"
`;

exports[`NotFoundPage > renders correctly 1`] = `
"<div data-v-fb0b02fd="" data-v-8f2b5bad="" data-v-c9c5b952="" class="py-10 px-8 vd-page-container d-flex justify-center">
  <p data-v-fb0b02fd="">Container size: 960</p>
  <div data-v-fb0b02fd="" class="v-sheet v-theme--light bg-transparent" style="width: 960px;">
    <div data-v-8f2b5bad="" class="v-card v-theme--light v-card--density-default elevation-0 v-card--variant-elevated pa-6 pa-sm-16">
      <!---->
      <div class="v-card__loader">
        <div class="v-progress-linear v-theme--light v-locale--is-ltr" style="top: 0px; height: 0px; --v-progress-linear-height: 2px;" role="progressbar" aria-hidden="true" aria-valuemin="0" aria-valuemax="100">
          <!---->
          <div class="v-progress-linear__background" style="opacity: NaN;"></div>
          <div class="v-progress-linear__buffer" style="opacity: NaN; width: 0%;"></div>
          <transition-stub name="fade-transition" appear="false" persisted="false" css="true">
            <div class="v-progress-linear__indeterminate">
              <div class="v-progress-linear__indeterminate long"></div>
              <div class="v-progress-linear__indeterminate short"></div>
            </div>
          </transition-stub>
          <!---->
        </div>
      </div>
      <!---->
      <!---->
      <div data-v-8f2b5bad="" class="v-row max-width-none">
        <div data-v-8f2b5bad="" class="v-col-sm-6 v-col-12 order-last order-sm-first text-center text-sm-left d-flex flex-column justify-center align-sm-start">
          <div data-v-8f2b5bad="" class="vd-code text-primary mb-4"><span data-v-8f2b5bad="" class="d-sr-only">Code d'erreur&nbsp;: </span> 404</div>
          <h2 data-v-8f2b5bad="" class="mb-2 font-weight-bold text-h5 mb-4">Page non trouvée</h2>
          <p data-v-8f2b5bad="">Cette page n’existe pas ou a été déplacée.</p>
          <!--v-if-->
        </div>
        <div data-v-8f2b5bad="" class="v-col-sm-6 v-col-12 d-flex align-center justify-center"><img data-v-c9c5b952="" src="/src/components/NotFoundPage/assets/not-found.svg" alt="Page non trouvée"></div>
      </div>
      <!---->
      <!----><span class="v-card__underlay"></span>
    </div>
  </div>
</div>"
`;
